bake/bake.go

```diff
@@ -67,29 +67,53 @@ func ReadLocalFiles(names []string) ([]File, error) {
 	return out, nil
 }
 
-func ReadTargets(ctx context.Context, files []File, targets, overrides []string, defaults map[string]string) (map[string]*Target, []*Group, error) {
+func ReadTargets(ctx context.Context, files []File, inp *Input, targets []string, overrides []string, defaults map[string]string) (map[string]*Target, map[string]*Group, map[string]build.Options, error) {
```

Comment:
What's the reason for changing this signature? Can't you just add the group setting and leave the rest of the function as is.

Reply:
Thought `TargetsToBuildOpt` had some impact on groups context but it's actually only against targets.

Reply:
I will bring back the `[]*Group` signature as before.

Reply:
Generally, I don't mind the map return but as this is going to be picked to release branch probably better to keep the change minimal. We can follow up with the type change in another PR that will not be picked later.